k8s/services/argocd/components/application.cue

```diff
@@ -31,36 +34,94 @@ import (
 			}
 			syncOptions: [...string] | *["CreateNamespace=true"]
 		}
-	}
-	k8sResources: "argocd-\(application.name)": resources.#Application & {
-		metadata: {
-			name:      application.name
-			namespace: k8s.namespace
+		credentials: {
+			privateKey?: v1.#Secret
+			externalSecret: {
+				refreshInterval: *"1h" | string
+				storeRef: {
+					name: string
+					kind: *"ClusterSecretStore" | "SecretStore"
+				}
+				decodingStrategy: *"None" | "Base64" | "Base64URL" | "Auto"
+			}
 		}
-		spec: {
-			destination: {
-				namespace: application.destination.namespace
-				server:    application.destination.server
+	}
+
+	convertGitHubURLToSSH: {
```

Comment:
the repo URL should be identical in the app and the secret, so this function is not needed